dLux/dev/jwst.py

```diff
@@ -1056,114 +478,6 @@ def _load(self : Layer, segment : str):
-    >>> second_slit = RectangularAperture(
-    ...     npix=1008, width=width, height=height, 
-    ...     x_offset = x_sep/2, y_offset=0.,
-    ...     theta=0., phi=0., magnification=0.)
-    >>> apertures = {"Right": first_slit, "Left": second_slit}
-    >>> double_slit = CompoundAperture(apertures)
-    >>> double_slit["Right"]
-
-    Attributes
-    ----------
-    npix : int
-        The number of pixels to be used to represent the entire 
-        set of apertures.
-    pixel_scale : float, meters per pixel
-        The length of the side of one pixel in the image. 
-    apertures : dict(str, Layer)
-        The apertures that make up the compound aperture. 
-    """
-    npix : int
-    pixel_scale : float
-    apertures : dict    
-
-
-    def __init__(self : Layer, number_of_pixels : int, 
-            pixel_scale : float, apertures : dict) -> Layer:
-        """
-        Parameters
-        ----------
-        number_of_pixels : int
-            The number of pixels used to represent the compound 
-            aperture. 
-        pixel_scale : float, meters per pixel
-            The length of one edge of a pixel in the rendered 
-            aperture.
-        apertures : dict
-            The aperture objects stored in a dictionary of type
-            {str : Layer} where the Layer is a subclass of the 
-            Aperture.
-        """
-        self.npix = int(number_of_pixels)
-        self.pixel_scale = float(pixel_scale)
-        self.apertures = apertures
-
-
-    def __getitem__(self : Layer, key : str) -> Layer:
-        """
-        Get one of the apertures from the collection using a name 
-        based lookup.
-        
-        Parameters
-        ----------
-        key : str
-            The name of the aperture to lookup. See the class doc
-            string for more information.
-        """
-        return self.apertures[key]
-
-
-    def __setitem__(self : Layer, key : str, value : Layer) -> None:
-        """
-        Assign a new value to one of the aperture mirrors.
-
-        Parameters
-        ----------
-        key : str
-            The name of the segement to replace for example "B1-7".
-        value : Layer
-            The new value to assign to that segement.
-        """
-        self.apertures[key] = value
-
-
-    def _aperture(self : Layer) -> Matrix:
-        """
-        Returns 
-        -------
-        aperture : Matrix
-            An aperture generated by combining all of the sub 
-            apertures that were stored. 
-        """
-        apertures = []
-        for aperture in self.apertures.values():
-            apertures.append(aperture._aperture())
-        return np.stack(apertures).sum(axis=0)
-
-
-    def get_npix(self : Layer) -> int:
-        """
-        Returns
-        -------
-        pixels : int
-            The number of pixels along one edge of the output image.
-        """
-        return self.npix
-
 
 class JWSTPrimaryAperture(CompoundAperture):
```

Comment:
Should this be in apertures.py?

Reply:
The hexagonal aperture was on the older generation and I did not get around to updating it so I didn't want to move it out of the `dev` folder yet.